src/prog_models/linear_model.py

```diff
@@ -0,0 +1,84 @@
+# Copyright © 2021 United States Government as represented by the Administrator of the
+# National Aeronautics and Space Administration.  All Rights Reserved.
+
+from . import PrognosticsModel
+from abc import ABC, abstractmethod
+import numpy as np
+
+
+class LinearModel(PrognosticsModel, ABC):
+    """
+    A linear prognostics model. Used when behavior can be described using a simple linear time-series model defined by the following equations:
+        * dx/dt = Ax + Bu + C
+        * z = Dx + E
+        * es = Fx + G
+    where x is state, u is input, z is output and es is event state
+
+    Linear Models must inherit from this class and define the following properties:
+        * A: 2-d numpy.array[float] 
```

Comment:
We could add matrix dimensions; makes life easier for the user

A -> n_states x n_states
B -> n_states x n_inputs
E -> n_states x n_state_disturbances 

C -> n_outputs x n_states
D -> n_outputs x n_meas_disturbances

F -> n_eventstates x n_states
G -> n_eventstates x n_es_disturbances